### ISSUE TITLE
Default density_floor warning does not trigger if density_floor absent from parameter file.

### ISSUE THREAD

**Issue by brantr** [cholla-hydro/cholla#372]
In global.cpp, there is a warning that looks intended to alert the user that they are using the default `density_floor=0`.  However, as written, this will only trigger if `density_floor` is defined in the parameter file:

```
#ifdef DENSITY_FLOOR
  } else if (strcmp(name, "density_floor") == 0) {
    parms->density_floor = atof(value);
    if (parms->density_floor == 0) {
      chprintf(
          "WARNING: density floor is set to its default value (zero)! It can be set to a different value in the input "
          "parameter file.\n");
    }
#endif
```

If `density_floor` doesn't appear in the parameter file, the` else if() `doesn't get executed.

I'm checking to see whether this was intended (warning the user they have set the density floor to zero), or whether a general warning about using `density_floor=0` was intended (regardless of the declaration of `density_floor` in the parameter file).

**Comment by brantr** [comment]
I have verified that setting `density_floor=0` in the parameter file triggers the warning.

**Comment by evaneschneider** [comment]
It should be the case that not setting the density floor parameter in the input file still triggers the warning, since it is given a default value of zero in the parameter struct header where it is declared. If not, we should fix it. The intended behavior is to warn the user if they have compiled with the density floor flag but haven’t set their desired floor in the input file. If this is unclear, we can work on the wording in the warning.

--
Evan Schneider
Assistant Professor of Physics and Astronomy
University of Pittsburgh
520-822-6294
pronouns: she/they
I support the Union of Pitt Faculty<https://www.pittfaculty.org/>

On Feb 6, 2024, at 5:23 PM, Brant Robertson ***@***.***> wrote:

﻿

I have verified that setting density_floor=0 in the parameter file triggers the warning.

**Comment by evaneschneider** [comment]
@helenarichie Can you take a look at this?

**Comment by evaneschneider** [comment]
Fixed in PR #373
